tests/libtest/lib1950.c

```diff
@@ -0,0 +1,541 @@
+/***************************************************************************
+ *                                  _   _ ____  _
+ *  Project                     ___| | | |  _ \| |
+ *                             / __| | | | |_) | |
+ *                            | (__| |_| |  _ <| |___
+ *                             \___|\___/|_| \_\_____|
+ *
+ * Copyright (C) 2013 - 2017, Linus Nielsen Feltzing, <linus@haxx.se>
+ *
+ * This software is licensed as described in the file COPYING, which
+ * you should have received as part of this distribution. The terms
+ * are also available at https://curl.haxx.se/docs/copyright.html.
+ *
+ * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+ * copies of the Software, and permit persons to whom the Software is
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ ***************************************************************************/
+#include "test.h"
+
+#include "testutil.h"
+#include "warnless.h"
+#include "memdebug.h"
+
+#define TEST_HANG_TIMEOUT 60 * 1000
+#define MAX_URLS 200
+#define MAX_BLACKLIST 20
+#define BUNDLE_SIZE 2
+#define NUM_HANDLE_REUSES 3
+
+#define CONTENT_LENGTH    "Content-Length: 8"
+#define DATAFMT           "%08.8u"
+#define DATALEN           8
+/*
+#define PADDINGLEN        256
+*/
+#define        MAX(a,b) (((a)>(b))?(a):(b))
+#define        MIN(a,b) (((a)<(b))?(a):(b))
+
+static int urltime[MAX_URLS];
+static char *urlstring[MAX_URLS];
+static CURL *handles[MAX_URLS];
+static char *site_blacklist[MAX_BLACKLIST];
+static char *server_blacklist[MAX_BLACKLIST];
+static int num_handles;
+static int blacklist_num_servers;
+static int blacklist_num_sites;
+static char *test_base_url;
+static struct curl_slist *curl_headers = NULL;
+static int write_req_sequence = 0;
+static int read_req_sequence = 0;
+
+static size_t
+write_callback(void *contents, size_t size, size_t nmemb, void *userp)
+{
+  size_t realsize = size * nmemb;
+  size_t datasize = MIN(DATALEN, realsize);
+/*
+  (void)contents;
+  (void)userp;
+*/
+  char buf[DATALEN + 1];
+
+  snprintf(buf, sizeof(buf), DATAFMT, (unsigned int) userp);
```

Comment:
compilers complain really loudly on this line. `userp` is a 64bit pointer that you typecast to a 32bit value here. Can I propose that you instead pass in a pointer to that integer and use a proper contents-of operator there instead?

Reply:
good point. all my testing was in a 32-bit environment.

Thank you for the patch. In addition to your changes, the pointer set with CURLOPT_WRITEDATA will need to be allocated somehow, otherwise by the time it is used it would be pointing to some invalid location on the stack. I will rework it.